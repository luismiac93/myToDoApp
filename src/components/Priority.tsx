import React from 'react';
import {Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {priorityName} from '../helpers/taskHelpers';
import {Task} from '../interfaces/task';

interface Props {
  task: Task;
}

export const Priority = ({task}: Props) => {
  return (
    <View style={{marginLeft: 40, flexDirection: 'row', marginTop: 5}}>
      <Icon
        name="flag-outline"
        color={
          task.priority === 2
            ? '#19196F'
            : task.priority === 1
            ? '#D9A443'
            : '#BD1302'
        }
        size={18}
      />
      <Text style={{marginLeft: 10}}>{priorityName(task.priority)}</Text>
    </View>
  );
};
