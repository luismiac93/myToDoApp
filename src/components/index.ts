export {Priority} from './Priority';
export {TaskCard} from './TaskCard';
export {Header} from './Header';
