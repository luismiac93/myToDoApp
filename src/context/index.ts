export {TaskProvider} from './task/TaskProvider';
export {TaskContext} from './task/TaskContext';
export {taskReducer} from './task/taskReducer';
