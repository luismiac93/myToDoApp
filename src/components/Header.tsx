import React, {useContext} from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {TaskContext} from '../context';
import {wereCompleted} from '../helpers';
import {styles} from '../theme/appTheme';

export const Header = () => {
  const {tasks, orderBy} = useContext(TaskContext);
  return (
    <View>
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'flex-start',
          marginTop: 10,
          marginBottom: 10,
        }}>
        <Icon name="list-outline" color={'whitesmoke'} size={35} />
        <Text style={styles.title}>My ToDo App</Text>
      </View>
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'center',
          marginTop: 10,
          marginBottom: 10,
        }}>
        <Text style={styles.subTitle}>Tasks: {tasks.length}</Text>
        <Text style={styles.subTitle}>Completed: {wereCompleted(tasks)}</Text>
      </View>
      {tasks.length > 0 && (
        <View
          style={{
            flexDirection: 'row',
            marginTop: 10,
            marginBottom: 10,
            justifyContent: 'space-around',
          }}>
          <TouchableOpacity
            onPress={() => {
              orderBy('name');
            }}>
            <View style={{flexDirection: 'row', marginRight: 10}}>
              <Icon name="filter-outline" color={'whitesmoke'} size={18} />
              <Text style={styles.subTitleFilters}>By name</Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => {
              orderBy('priority');
            }}>
            <View style={{flexDirection: 'row'}}>
              <Icon name="filter-outline" color={'whitesmoke'} size={18} />
              <Text style={styles.subTitleFilters}>By priority</Text>
            </View>
          </TouchableOpacity>
        </View>
      )}
    </View>
  );
};
