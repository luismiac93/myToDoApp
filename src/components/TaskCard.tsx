import React, {useContext, useState} from 'react';
import {Task} from '../interfaces/task';
import {Text, TouchableOpacity, View} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {TaskContext} from '../context';
import BouncyCheckbox from 'react-native-bouncy-checkbox';
import {Priority} from './Priority';

interface Props {
  task: Task;
}

export const TaskCard = ({task}: Props) => {
  const {deleteTask, completedTask} = useContext(TaskContext);
  const [checked, setChecked] = useState(task.status);

  const deleteCurrentTask = (id: string) => {
    deleteTask(id);
  };

  const completedCurrentTask = (id: string) => {
    setChecked(!checked);
    completedTask(id);
  };

  return (
    <View
      style={{
        backgroundColor: 'whitesmoke',
        borderRadius: 10,
        marginBottom: 10,
        padding: 10,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
      }}>
      <View>
        <BouncyCheckbox
          isChecked={checked}
          text={task.task}
          textStyle={{color: '#000000', fontWeight: 'bold'}}
          disableBuiltInState
          onPress={() => completedCurrentTask(task.id)}
        />
        <Priority task={task} />
      </View>
      <TouchableOpacity onPress={() => deleteCurrentTask(task.id)}>
        <Icon name="trash-outline" color={'#19196F'} size={22} />
      </TouchableOpacity>
    </View>
  );
};
