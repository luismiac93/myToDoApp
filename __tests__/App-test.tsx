/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {Header} from '../src/components/Header';
import {TaskContext} from '../src/context';
import {Priority} from '../src/components/Priority';

test('Header render correctly', () => {
  const tree = renderer
    .create(
      <TaskContext.Provider
        value={{
          tasks: [],
          completedTask: () => {},
          createTask: () => {},
          deleteTask: () => {},
          orderBy: () => {},
        }}>
        <Header />
      </TaskContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Priority render correctly', () => {
  const tree = renderer
    .create(
      <Priority
        task={{id: '', priority: 0, task: 'do some test', status: false}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
