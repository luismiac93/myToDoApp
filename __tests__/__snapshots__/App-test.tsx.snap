// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "whitesmoke",
            "fontSize": 35,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginLeft": 10,
        }
      }
    >
      My ToDo App
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": 16,
          "fontWeight": "600",
          "marginLeft": 10,
        }
      }
    >
      Tasks: 
      0
    </Text>
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": 16,
          "fontWeight": "600",
          "marginLeft": 10,
        }
      }
    >
      Completed: 
      0
    </Text>
  </View>
</View>
`;

exports[`Priority render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginLeft": 40,
      "marginTop": 5,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#BD1302",
          "fontSize": 18,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    High Priority
  </Text>
</View>
`;
